app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx

```diff
@@ -167,67 +203,88 @@ const TransactionDetails: React.FC<Props> = ({ order }: Props) => {
 					<ListItem.Content style={styles.listItems}>
 						<ListItem.Body>
 							<Text black small>
-								Payment Method
+								{strings('fiat_on_ramp_aggregator.transaction.payment_method')}
 							</Text>
 						</ListItem.Body>
 						<ListItem.Amount>
 							<Text small bold primary>
-								{order.data.paymentMethod.name}
+								{order.paymentMethod?.name || data.paymentMethod?.name}
 							</Text>
 						</ListItem.Amount>
 					</ListItem.Content>
 					<Text small style={styles.provider}>
-						{strings('fiat_on_ramp_aggregator.transaction.via')} {order.data.provider.name}
+						{strings('fiat_on_ramp_aggregator.transaction.via')}{' '}
+						{order.provider?.name || data.provider?.name}
```

Comment:
`order. provider` is a `string`,  use [`getProviderName`](https://github.com/MetaMask/metamask-mobile/blob/e632be20eac704ee6bbfd9a0713d0e3f49e54c43/app/reducers/fiatOrders/index.js#L51)  passing `data` to get the provider name

Reply:
For now yes, we can change it later in `getProviderName` and it will be updated in every occurrence